app/controllers/requests.rb

```diff
@@ -0,0 +1,89 @@
+class MakersBnb < Sinatra::Base
+
+  get '/requests' do
+    if current_user
+      @users_requests = Request.all(user_id: current_user.id)
+      @received_requests = Request.all(owner_id: current_user.id)
+      @request_display = @users_requests.map do |x|
+        Space.get(x.space_id)
+      end
+      erb :'requests/index'
+    else
+      redirect to '/sessions/new'
+    end
+  end
+
+  post '/requests/new' do
+
+    requested_date = Date.parse(params[:date_requested]) rescue nil
```

Comment:
That sounds like a plan. I'd start by writing how I wish the code could look, then finding a solution to make it work. That is: define the problem tightly, then look for a solution to that small problem. For me, that would be something like this:

```ruby
post '/requests/new' do
  date_requested = SanitizedDate.new(date_requested)
  return redirect '/spaces', notice: 'Please enter a valid date' unless date_requested

  space = Space.get(params[:space_id])
  return redirect '/spaces', notice: 'Space not available on that date' unless space.available_on?(date_requested)

  redirect '/spaces'
end
```

Although I feel there's better design in there somewhere, because that's 5 lines.